packages/doenetml-worker-rust/lib-rust-macros/src/component_node.rs

```diff
@@ -4,14 +4,28 @@ use proc_macro2::{Ident, Span};
 use quote::quote;
 use syn::{self, FieldsNamed};
 
+use darling::FromDeriveInput;
+
 use crate::util::has_attribute;
 
+/// Extra options a component may pass to the `ComponentNode` derive macro.
+#[derive(Debug, Default, FromDeriveInput)]
+#[darling(attributes(component))]
+pub struct ComponentOptions {
+    /// Accessible with `#[component(ref_transmutes_to = "new_name")]`.
+    /// Indicates what component a reference should become. For example
+    /// `<textInput name="a"/>$a`, one may want `$a` to appear as a `Text` by default,
+    /// rather than a `TextInput`. In this case, you would set `ref_transmutes_to = "Text"`.
+    pub ref_transmutes_to: Option<String>,
+}
+
 /// Implement the ComponentNode trait structs
 /// assuming they have have a common field that is `ComponentCommonData`
 pub fn component_node_derive(input: TokenStream) -> TokenStream {
     let ast: syn::DeriveInput = syn::parse(input).unwrap();
     let name = &ast.ident;
     let data = &ast.data;
+    let options: ComponentOptions = FromDeriveInput::from_derive_input(&ast).unwrap();
```

Comment:
Nice, you found a much better way to get the attributes for a derive macro!

Reply:
Yeah. The `darling` package looks pretty cool :-)